crmprtd/download_cache_process.py

```diff
@@ -0,0 +1,371 @@
+            connection_string=connection_string,
+        )
+    else:
+        now = datetime.datetime.now()
+        download_and_process(
+            network_name=network_name,
+            log_args=log_args(**kwargs),
+            download_args=download_args(**kwargs, time=now),
+            cache_filename=cache_filename(**kwargs, timestamp=now),
+            connection_string=connection_string,
+        )
+
+
+def dispatch_network_alias(network_alias: str = None, **kwargs) -> None:
+    """
+    Dispatch each network defined by an alias.
+
+    :param network_alias: Alias name.
+    :param kwargs: Remaining args, passed through.
+    :return: None.
+    """
+    assert network_alias in network_alias_names
+    for network_name in alias_to_networks(network_alias):
+        dispatch_network(network_name=network_name, **kwargs)
+
+
+def dispatch(network: str = None, **kwargs) -> None:
+    """
+    Dispatch a network or a network alias.
+
+    :param network: Network name or network alias name.
+    :param kwargs: Remain args, passed through.
+    :return: None.
+    """
+    assert network is not None
+    if network in network_alias_names:
+        dispatch_network_alias(network_alias=network, **kwargs)
+    elif network in NETWORKS:
+        dispatch_network(network_name=network, **kwargs)
+    else:
+        raise ValueError(
+            f"Network argument '{network}' is not a valid network name or alias"
+        )
+
+
+def describe_network(network):
+    """Return a string describing `network`, which may be a network name or alias."""
+    if network in NETWORKS:
+        return f"'{network}' is the name of a single network"
+    if network in network_alias_names:
+        return (
+            f"'{network}' is the name of a network alias, which translates to the "
+            f"following network names: {', '.join(network_aliases[network])}"
+        )
+    return f"'{network}' is not a known network name or alias"
+
+
+def main(arglist: List[str] = None) -> None:
+    """
+    Mainline for script.
+
+    Note: This script allows for only two options:
+    1. Download data and cache only (omit database connection string).
+    2. Download data, cache it, and process it.
+
+    :param arglist: Argument list to be processed by argparse. This eases testing;
+        if None then sys.argv is used.
+    :return: None.
+    """
+
+    parser = ArgumentParser(
+        description="""
+            The download-cache-process dispatcher. Starts two subprocesses
+            running crmprtd_download and crmprtd_process with appropriate arguments,
+            pipes the first into the second, and caches the downloaded data.
+        """
+    )
+
+    add_version_arg(parser)
+
+    parser.add_argument(
+        "-D",
+        "--describe",
+        nargs=1,
+        choices=NETWORKS + network_alias_names,
+        action=OneAndDoneAction,
+        function=describe_network,
+        help="Describe a network name or alias",
+    )
```

Comment:
This option should be pulled out into a separate script, `crmprtd_describe`

Reply:
On second thought, this should stay where it is, as it includes aliases, which are only valid with respect to `download_cache_process`